core/templates/pages/admin-page/admin-page.component.spec.ts

```diff
@@ -0,0 +1,142 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview UnitTests for Admin Page component.
+ */
+
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ChangeDetectorRef, NO_ERRORS_SCHEMA } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { PlatformFeatureService } from 'services/platform-feature.service';
+import { AdminPageComponent } from './admin-page.component';
+import { AdminRouterService } from './services/admin-router.service';
+
+class MockWindowRef {
+  nativeWindow = {
+    confirm() {
+      return true;
+    },
+    location: {
+      hostname: 'hostname',
+      href: 'href',
+      pathname: 'pathname',
+      search: 'search',
+      hash: 'hash'
+    },
+    open() {
+      return;
+    },
+    onhashchange() {
+      return;
+    }
+  };
+}
+
+describe('Admin misc tab component ', () => {
+  let component: AdminPageComponent;
+  let fixture: ComponentFixture<AdminPageComponent>;
+
+  let adminRouterService: AdminRouterService;
+  let platformFeatureService: PlatformFeatureService;
+  let mockWindowRef: MockWindowRef;
+
+  beforeEach(() => {
+    mockWindowRef = new MockWindowRef();
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ],
+      declarations: [AdminPageComponent],
+      providers: [
+        AdminRouterService,
+        ChangeDetectorRef,
+        PlatformFeatureService,
+        {
+          provide: WindowRef,
+          useValue: mockWindowRef
+        }
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(AdminPageComponent);
+    component = fixture.componentInstance;
+  });
+
+  beforeEach(() => {
+    adminRouterService = TestBed.inject(AdminRouterService);
+    platformFeatureService = TestBed.inject(PlatformFeatureService);
+
+    spyOn(adminRouterService, 'showTab').and.returnValue(null);
+  });
+
+  it('should check whether the admin activities tab is open', () => {
+    // Setting admin activities tab to be open.
+    spyOn(adminRouterService, 'isActivitiesTabOpen').and.returnValue(true);
+    let result = component.isActivitiesTabOpen();
+    expect(result).toBe(true);
+  });
+
+  it('should check whether the admin config tab is open', () => {
+    // Setting admin config tab to be open.
+    spyOn(adminRouterService, 'isConfigTabOpen').and.returnValue(true);
+    let result = component.isConfigTabOpen();
+    expect(result).toBe(true);
```

Comment:
Can we test it like this? Same for the rest of the tests. You can do it if you want.

```suggestion
    expect(component.isConfigTabOpen()).toBe(true);
```

Reply:
I think it's just preference